src/classes/frModel.cls

```diff
@@ -0,0 +1,71 @@
+public abstract class frModel {
+	protected abstract SObject getObject();
+    public abstract List<frMapping__c> getMappings();
+
+    public virtual void populateFromRequest(Map<String, Object> request) {
+    	Map<String, Schema.SObjectField> fields = frSchemaUtil.getFields(getObject().getSObjectType().getDescribe().getName());
+
+        Map<String, frMapping__c> frNameToSfName = new Map<String, frMapping__c>();
+        for(frMapping__c mapping : getMappings()) {
+            frNameToSfName.put(mapping.fr_Name__c, mapping);
+        }
+
+        for(String fieldName : request.keySet()) {
+            if(frNameToSfName.containsKey(fieldName)) {
+                frMapping__c mapping = frNameToSfName.get(fieldName);
+                Object value = mapping.Is_Constant__c ? mapping.Constant_Value__c : request.get(fieldName);
+                write(getObject(), fields, mapping.sf_Name__c, value);
+            }
+        }
+        getObject().put('fr_ID__c', String.valueOf(request.get('id')));
+    }
+
+    private static void write(SObject record, Map<String, Schema.SObjectField> fields, String fieldName, Object value) {
```

Comment:
@Alexalex1432 what do you think about adding some `handleExceptions()` here to be more verbose about when field writes fail?